app/furniture/marketplace/routes.rb

```diff
@@ -0,0 +1,14 @@
+class Marketplace
+  class Routes
+    def self.append_routes(router)
+      router.resources :marketplaces, only: [:show, :edit, :update], module: "marketplace" do
+        router.resources :orders, only: [:show]
+        router.resources :products
+        router.resources :carts do
+          router.resources :cart_products
+        end
+        router.resources :checkouts, only: [:show, :create]
+      end
+    end
+  end
+end
```

Comment:
why does `Marketplace` get it's own `routes.rb` file? Why not just put it in
https://github.com/zinc-collective/convene/blob/main/config/routes.rb

Reply:
A few reasons:

1. If we put in in `config/routes.rb` then every developer maintaining a piece of  `Furniture` would need to modify `routes.rb` in order to adjust the routes their `Furniture` users. Instead, we hook in to the `Furniture` class in `config/routes.rb` and tell it to iterate through every piece of furniture, and append the routes defined within the `Furniture` itself.

This allows `Furniture` pieces to "own" the code related to their furniture within the `app/furniture/furniture_name` folder; which reduces the amount of surface area a Furniture maintainer needs to cover in order to make significant changes to their `Furniture`'s behavior. 

TL/DR: Encapsulation! Every piece of Furniture owns it's entire implementation, from Routes to Models to Controllers to Policies to Views; within the `app/furniture/furniture_name` directory.